packages/actor-rdf-parse-html-script/lib/ActorRdfParseHtmlScript.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {ActionContext, Actor, Mediator} from "@comunica/core";
+import {Readable} from "stream";
+
+/**
+ * A HTML RDF Parse actor that listens on the 'rdf-parse' bus.
+ *
+ * It is able to extract JSON-LD from HTML files and parse the JSON-LD based RDF serializations
+ * and announce the presence of them by media type.
+ */
+export class ActorRdfParseHtmlScript extends ActorRdfParseFixedMediaTypes {
+
+  public mediatorRdfParse: Mediator<Actor<IActionRootRdfParse, IActorTestRootRdfParse,
+    IActorOutputRootRdfParse>, IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+    Promise<IActorRdfParseOutput> {
+
+    const myQuads = new Readable({objectMode: true});
+
+    let initialized = false;
+
+    myQuads._read = async () => {
+      if (!initialized) {
+        initialized = true;
+
+        // Stringify HTML input
+        const htmlString: string = await require('stream-to-string')(action.input);
+
+        // Extract script tags
+        const DOMParser = require('xmldom').DOMParser;
+        const doc = new DOMParser().parseFromString(htmlString, 'text/html');
+        const scripts = doc.getElementsByTagName('script');
+
+        const supportedTypes = [
```

Comment:
Hardcoding things like this is the root of all evil ;-)

You can dynamically discover all available content types by doing something like this:
```javascript
Object.keys((await this.mediatorRdfParse.mediate({context, mediaTypes: true})).mediaTypes);
```

Reply:
application/json+ld is not included in these mediaTypes. Where do I add it? Also, shouldn't text/html also be added?

Reply:
> application/json+ld is not included in these mediaTypes

It should be there though. Which ones _are_ included then?

> shouldn't text/html also be added?

No. You don't want script tags with type `text/html`.

Reply:
```
[ 'application/trig',
  'application/n-quads',
  'text/turtle',
  'application/n-triples',
  'text/n3' ]
```

Reply:
Also, running my tests with `Object.keys(...` results in a null value.

Reply:
Ah, right, that's because your mediator only picks a single actor.

You'll need another mediator over the same bus, but of type `mediator-combine-union`. You should be able to do this like in `ActorRdfDereferenceHttpParse`. (Can you double check if the JSON-LD content types are present in `ActorRdfDereferenceHttpParse`? Because I can't find usage of `mediator-combine-union` in the config file. Otherwise, it might be wrong there as well...)

Reply:
> Also, running my tests with Object.keys(... results in a null value.

That's because your test environment doesn't have any parsing actors registered.

Reply:
The JSON-LD content types are not in `ActorRdfDereferenceHttpParse` either.

Reply:
Ok, thanks for checking, reported here: #318.

Feel free to look into it, as I have no idea if or when I'll be able to look into it myself.